### DISCUSSION TITLE
Difference between two unsubscribe methods?

### DISCUSSION THREAD

**Discussion by aonghas** [pocketbase/pocketbase#5764] [Q&A]
In the docs, I see:

> pb.collection('example').unsubscribe('\*'); // remove all '*' topic subscriptions
> pb.collection('example').unsubscribe(); // remove all subscriptions in the collection

What's the difference? What is a '*' topic?

**Comment by ganigeorgiev** [comment]
A realtime subscription topic is the unique identifier of a subscription (_"channel" is another common name for this type of things_).

For the collections subscriptions the topic could be either a single record id (_in this case the View API rule would be used for access checks_) or `*` for when you want to listen to changes to all records from the collection (_in this case the List API rule would be used for access checks_).

In a single application it is allowed to have combination of both depending on your specific use case and where the subscription happen (a single instance of the JS SDK tries to maintain a single realtime connection no matter of how many `subscribe` calls you have).

I'm not sure how to explain it better but I'll try with an example:

```js
pb.collection("example").subscribe("RECORD_ID1", (e) => { ... }) // A
pb.collection("example").subscribe("RECORD_ID1", (e) => { ... }) // B
pb.collection("example").subscribe("RECORD_ID2", (e) => { ... }) // C
pb.collection("example").subscribe("*", (e) => { ... }) // D
pb.collection("example").subscribe("*", (e) => { ... }) // E
pb.collection("somethingElse").subscribe("*", (e) => { ... }) // F
```

Calling `pb.collection("example").unsubscribe("RECORD_ID1")` should remove only `A` and `B` subscriptions.

Calling `pb.collection("example").unsubscribe("*")` should remove only `D` and `E` subscriptions.

Calling `pb.collection("example").unsubscribe()` should remove `A`, `B`, `C`, `D`, `E` subscriptions.

**Reply by aonghas** [reply]
Great that makes sense! Is there a way to unsubscribe to all individual records in one call without keeping track of which records you've subscribed to? i.e unsubscribe to `A`, `B` and `C` only?

**Reply by ganigeorgiev** [reply]
No, there is no such option at the moment and I'm reluctant to add it as I don't understand the use case for it and don't want to complicate further the existing APIs unnecessary.

One possible workaround if you really need to do this could be to call `unsubscribe()` and then immediately resubscribe to the `*` with `subscribe("*", ...)`.
